AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin.swift

```diff
@@ -76,7 +76,11 @@ final public class AWSDataStorePlugin: DataStoreCategoryPlugin {
             return
         }
 
-        storageEngine = try StorageEngine(isSyncEnabled: isSyncEnabled)
+        storageEngine = try StorageEngine(isSyncEnabled: isSyncEnabled, dataStorePublisher: resolveDataStorePublisher())
+    }
+
+    private func resolveDataStorePublisher() -> DataStorePublisherBehavior? {
```

Comment:
It's confusing that this method returns the `dataStorePublisher` property, but that property is an instance of `DataStoreSubscribeBehavior`. Do these need to be combined, or separated, or renamed?